src/topology/unit_test.rs

```diff
@@ -22,30 +23,55 @@ pub struct UnitTestTransform {
 pub struct UnitTest {
     pub name: String,
     input: (String, Event),
-    transforms: HashMap<String, UnitTestTransform>,
+    transforms: IndexMap<String, UnitTestTransform>,
     checks: Vec<UnitTestCheck>,
 }
 
 //------------------------------------------------------------------------------
 
-fn event_to_string(event: Event) -> String {
+fn event_to_string(event: &Event) -> String {
     match event {
-        Event::Log(log) => serde_json::to_string(&log.unflatten()).unwrap_or_else(|_| "{}".into()),
+        Event::Log(log) => {
+            serde_json::to_string(&log.clone().unflatten()).unwrap_or_else(|_| "{}".into())
+        }
         Event::Metric(metric) => serde_json::to_string(&metric).unwrap_or_else(|_| "{}".into()),
     }
 }
 
+fn events_to_string(name: &str, events: &Vec<Event>) -> String {
```

Comment:
In `UnitTest::run` this is called like so:

https://github.com/timberio/vector/blob/5addaea1eb820cc6420977d55f20ca43dd331768/src/topology/unit_test.rs#L107-L114

https://github.com/timberio/vector/blob/5addaea1eb820cc6420977d55f20ca43dd331768/src/topology/unit_test.rs#L136-L144

It seems like this function is essentially trying to manual encode JSON here. Is this because you're using `event_to_string` to call the JSON encoding on the events in the vector? Could you call `serde_json::to_string(events.iter().map(|e| event_to_string(e)).collect::<Vec<_>>()` and let `serde_json` handle the the list formatting?

Reply:
It's a little more than that, when the number of events is 1 we print a single object on one line:

```
  input: {"foo":"bar"}
```

And when the events are > 1 we print an event on each line:

```
  inputs:
    {"foo":"bar"}
    {"foo":"baz"}
```

Which I think is worth it, as it's clearer and easier to read than:

```
  inputs: [{"foo":"bar"}]
```

And:

```
  inputs: [{"foo":"bar"},{"bar":"baz"}]
```

Reply:
Hmm, roger. I understand. :) Do you think it's OK that this kind of breaks machine readability? Maybe we could use some sort of prettyprinted JSON instead?